static/src/javascripts/projects/commercial/modules/header-bidding/prebid/prebid.js

```diff
@@ -136,6 +136,23 @@ const initialise = (window, framework = 'tcfv2') => {
         };
     }
 
+    if(config.get('switches.prebidImproveDigital', false)) {
+        // Add placement ID for Improve Digital, reading from the bid response
+        const REGEX_PID = new RegExp(/placement_id=\\?\"(\d+)\\?\"/)
+        window.pbjs.bidderSettings.improvedigital = {
+            adserverTargeting: [
+                {
+                    key: 'hb_pid',
+                    val(bidResponse) {
+                        const matches = REGEX_PID.exec(bidResponse.ad);
+                        const pid = matches && matches[1];
+                        return pid ?? '';
```

Comment:
Question: from what I see here, if there are no matches, `hb_pid` is set to an empty string. Would it make sense to do this check before, so the `window.pbjs.bidderSettings.improvedigital` object is never declared if `pid` is `null`?

Reply:
That’s an excellent point, there’s also a `suppressEmptyKeys` boolean that can be set:

```suggestion
                        const matches = REGEX_PID.exec(bidResponse.ad);
                        const pid = matches && matches[1];
                        return pid ?? null;
```